airbyte-cdk/python/airbyte_cdk/sources/file_based/schema_helpers.py

```diff
@@ -166,29 +167,29 @@ def conforms_to_schema(record: Mapping[str, Any], schema: Mapping[str, Any]) ->
 
     The record conforms to the supplied schema iff:
     - All columns in the record are in the schema.
-    - For every column in the record, that column's type is equal to or narrower than the same column's
-      type in the schema.
+    - For every column in the record, that column's type is equal to or narrower than the same column's type in the schema.
     """
-    schema_columns = set(schema.get("properties", {}).keys())
-    record_columns = set(record.keys())
+    schema_properties = schema.get("properties", {})
+    schema_columns = schema_properties.keys()
 
-    if not record_columns.issubset(schema_columns):
+    if not record.keys() <= schema_columns:
         return False
 
-    for column, definition in schema.get("properties", {}).items():
+    for column, definition in schema_properties.items():
         expected_type = definition.get("type")
         value = record.get(column)
 
         if value is not None:
             if isinstance(expected_type, list):
-                return any(is_equal_or_narrower_type(value, e) for e in expected_type)
+                if not any(is_equal_or_narrower_type(value, e) for e in expected_type):
```

Comment:
this doesn't seem exactly equivalent: on master, we'll always return whereas this version will keep iterating over the columns.

the new behavior seems right, but we should submit it separately with a corresponding unit test